ark/utils/notebooks_test_utils.py

```diff
@@ -0,0 +1,224 @@
+                    xr_channel_names=None, force_ints=True):
+    """Processes and saves the generated segmentation labels and runs the summed channel overlay
+
+    Args:
+        tb (testbook.testbook):
+            The testbook runner instance
+        delimiter (str):
+            The suffix of the files to read from deepcell_output_dir
+        xr_dim_name (str):
+            The dimension containing the channel names to read in
+        xr_channel_names (list):
+            The list of the channels we wish to read in
+        force_ints (bool):
+            Whether to convert the segmentation labels to integer type
+    """
+
+    delimiter_str = delimiter if delimiter is not None else "None"
+    xr_channel_names_str = str(xr_channel_names) if xr_channel_names is not None else "None"
+
+    # load the segmentation label with the proper command
+    load_seg_cmd = """
+        segmentation_labels = load_utils.load_imgs_from_dir(
+            data_dir=deepcell_output_dir,
+            delimiter="%s",
+            xr_dim_name="%s",
+            xr_channel_names=%s,
+            force_ints=%s
+        )
+    """ % (delimiter_str,
+           xr_dim_name,
+           xr_channel_names,
+           str(force_ints))
+    tb.inject(load_seg_cmd, after='load_seg_labels')
+
+    tb.execute_cell('save_seg_labels')
+
+    # now overlay data_xr
+    tb.execute_cell('load_summed')
+    tb.execute_cell('overlay_mask')
+
+
+def create_exp_mat(tb, is_mibitiff=False, batch_size=5):
+    """Creates the expression matrices from the generated segmentation labels
+
+    Args:
+        tb (testbook.testbook):
+            The testbook runner instance
+        is_mibitiff (bool):
+            Whether we're working with mibitiff images
+        batch_size (int):
+            The number of fovs we want to process at a time
+    """
+
+    exp_mat_gen = """
+        cell_table_size_normalized, cell_table_arcsinh_transformed = \
+            marker_quantification.generate_cell_table(segmentation_labels=segmentation_labels,
+                                                      tiff_dir=tiff_dir,
+                                                      img_sub_folder="TIFs",
+                                                      is_mibitiff=%s,
+                                                      fovs=fovs,
+                                                      batch_size=%s)
+    """ % (is_mibitiff, str(batch_size))
+    tb.inject(exp_mat_gen)
+
+    # save expression matrix
+    tb.execute_cell('save_exp_mat')
+
+
+def remove_dirs(tb):
+    """Removes all of the test folders created
+
+    Args:
+        tb (testbook.testbook):
+            The testbook runner instance
+    """
+
+    remove_dirs = """
+        from shutil import rmtree
+        rmtree(tiff_dir)
+        rmtree(deepcell_input_dir)
+        rmtree(deepcell_output_dir)
+        rmtree(single_cell_dir)
+    """
+    tb.inject(remove_dirs)
```

Comment:
weird nitpick, but I'm thinking that temp folder creation + clean up should maybe happen outside of the notebook.

I think all that would need to happen, is to skip the execution of the 'create_dirs' cell, make all the directories from within a tempfolder, and then let python do the clean up afterwards.

The only reason I bring this up, is because if someone runs this test locally, and has data in folders whose names happen to match those of the test folders, all that data would be deleted after the test runs.

Reply:
Great suggestion, I'll put that in!